### ISSUE TITLE
Add an Agent Skills (SKILL.md) validation ruleset

### ISSUE THREAD

**Issue by joshrotenberg** [joshrotenberg/mdbook-lint#458]
## Summary

Add an optional Agent Skills ruleset for validating `SKILL.md` files against the open [Agent Skills specification](https://agentskills.io/specification).

This is a strong fit for the modular provider architecture: unlike ordinary prose conventions, Agent Skills has a normative filename, YAML frontmatter schema, field constraints, directory-name relationship, and progressive-disclosure guidance.

Hyalo has independently added a `skills` profile for this use case, which is useful validation that users want Markdown linting and skill-schema validation in one tool.

## Why this belongs here

- `SKILL.md` is Markdown with a small, well-defined structured preamble.
- The existing ADR provider establishes the pattern for format-specific, feature-gated rulesets with YAML parsing.
- Standard MD rules can lint the body while a dedicated provider validates skill semantics.
- Collection-aware/path-aware rules can validate the parent-directory relationship and local references.
- This broadens mdbook-lint from mdBook-specific linting into a reusable Markdown rules platform without weakening existing rule namespaces.

## Proposed provider

```rust
AgentSkillRuleProvider
```

Suggested Cargo feature:

```toml
skills = ["dep:serde_yaml"]
```

The provider should apply only to files named exactly `SKILL.md`.

## Proposed MVP rules

The final IDs/names are open to bikeshedding, but an initial catalog could be:

| ID | Check | Default |
|---|---|---|
| SKILL001 | Required, parseable YAML frontmatter | error |
| SKILL002 | Required `name`; 1–64 chars; lowercase ASCII letters/numbers/hyphens; no leading, trailing, or consecutive hyphens | error |
| SKILL003 | `name` matches the parent directory | error or warn, matching spec/reference-validator behavior |
| SKILL004 | Required non-empty `description`, maximum 1024 characters | error |
| SKILL005 | Optional field types and bounds: `compatibility` 1–500 chars, `metadata` string map, `allowed-tools` string | error |
| SKILL006 | Main file exceeds the recommended 500-line budget | warn |

A later advisory rule may validate relative file references and warn on deeply nested reference chains, but that should not block the frontmatter MVP.

## Specification notes

The official specification defines:

- exact `SKILL.md` naming and directory structure;
- required `name` and `description`;
- name syntax, length, and parent-directory match;
- optional `license`, `compatibility`, `metadata`, and experimental `allowed-tools`;
- a recommended limit of 500 lines / approximately 5,000 tokens for the main instructions;
- relative, shallow references for progressive disclosure.

The official `skills-ref validate` implementation should be used as a compatibility oracle where practical.

## Acceptance criteria

- [ ] An optional `skills` feature exposes `AgentSkillRuleProvider`.
- [ ] Only exact `SKILL.md` paths activate the provider.
- [ ] Normative frontmatter failures are errors; recommendations are warnings.
- [ ] Valid and invalid fixtures cover every field and boundary length.
- [ ] Directory-name matching works for nested skill collections.
- [ ] The body continues through the ordinary standard Markdown rules.
- [ ] Results are compared with `skills-ref validate` on a shared fixture corpus.
- [ ] CLI/library enabling behavior and rule IDs are documented.
- [ ] The implementation does not impose product-specific Claude/Codex conventions that are absent from the open specification.

References:

- [Agent Skills specification](https://agentskills.io/specification)
- [Reference implementation repository](https://github.com/agentskills/agentskills/tree/main/skills-ref)
- [Hyalo skills profile overview](https://github.com/ractive/hyalo#profiles)
